xarray/core/computation.py

```diff
@@ -992,7 +1019,7 @@ def earth_mover_distance(first_samples,
     if kwargs is None:
         kwargs = {}
 
-    signature = _UFuncSignature(input_core_dims, output_core_dims)
+    signature = _UFuncSignature(input_core_dims, output_core_dims, exclude_dims)
```

Comment:
As far as I can tell, we only need `exclude_dims` inside the `to_gufunc_string` method. What about just adding it as an optional argument there?